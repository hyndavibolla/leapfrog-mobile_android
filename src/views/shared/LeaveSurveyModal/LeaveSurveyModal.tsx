import React, { memo } from 'react';
import { View } from 'react-native';

import { styles } from './styles';
import { useTestingHelper } from '../../../utils/useTestingHelper';
import { Title, TitleType } from '../Title';
import { Button } from '../Button';
import ExclamationCircle from '../../../assets/shared/exclamationCircle.svg';

export interface Props {
  onLeaveSurvey?: () => void;
  onCancel?: () => void;
}

export const LeaveSurveyModal = ({ onLeaveSurvey, onCancel }: Props) => {
  const { getTestIdProps } = useTestingHelper('leave-survey-modal');
  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <ExclamationCircle style={styles.swyLogo as any} />
      <Title style={styles.title} type={TitleType.SECTION}>
        Do you want to leave Surveys?
      </Title>
      <Button innerContainerStyle={styles.updateBtnInnerContainer} textStyle={styles.updateBtnText} onPress={onLeaveSurvey} {...getTestIdProps('leave-btn')}>
        Yes, leave Surveys
      </Button>
      <Button innerContainerStyle={styles.dismissBtnInnerContainer} textStyle={styles.dismissBtnText} onPress={onCancel} {...getTestIdProps('cancel-btn')}>
        Cancel
      </Button>
    </View>
  );
};

export default memo(LeaveSurveyModal);
