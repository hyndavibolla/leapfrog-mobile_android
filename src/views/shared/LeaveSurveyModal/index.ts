export { default as LeaveSurveyModal } from './LeaveSurveyModal';
