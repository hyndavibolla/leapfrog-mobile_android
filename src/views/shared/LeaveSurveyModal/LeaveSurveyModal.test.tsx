import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import LeaveSurveyModal, { Props } from './LeaveSurveyModal';

describe('LeaveSurveyModal', () => {
  let props: Props;
  beforeEach(() => {
    props = {
      onLeaveSurvey: jest.fn(),
      onCancel: jest.fn()
    };
  });

  it('should render', () => {
    const { toJSON } = render(<LeaveSurveyModal {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a pressable leave survey button', () => {
    const { getByTestId } = render(<LeaveSurveyModal {...props} />);
    fireEvent.press(getByTestId('leave-survey-modal-leave-btn'));
    expect(props.onLeaveSurvey).toBeCalled();
  });

  it('should have a pressable cancel button', () => {
    const { getByTestId } = render(<LeaveSurveyModal {...props} />);
    fireEvent.press(getByTestId('leave-survey-modal-cancel-btn'));
    expect(props.onCancel).toBeCalled();
  });
});
