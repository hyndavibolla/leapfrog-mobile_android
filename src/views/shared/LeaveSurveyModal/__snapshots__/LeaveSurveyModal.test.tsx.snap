// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveSurveyModal should render 1`] = `
<View
  accessibilityLabel="leave-survey-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="leave-survey-modal-container"
>
  ExclamationCircle
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "marginBottom": 32,
          "textAlign": "center",
        },
      ]
    }
    type="sectionHeader"
  >
    Do you want to leave Surveys?
  </Text>
  <View
    accessibilityLabel="leave-survey-modal-leave-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="leave-survey-modal-leave-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "paddingBottom": 16,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "SFProDisplay-Heavy",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Yes, leave Surveys
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="leave-survey-modal-cancel-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="leave-survey-modal-cancel-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "color": "#0066CC",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
